packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -691,6 +699,17 @@ export interface SdkPackage<TServiceOperation extends SdkServiceOperation> {
   enums: SdkEnumType[];
   unions: (SdkUnionType | SdkNullableType)[];
   crossLanguagePackageId: string;
+  namespaces: SdkNamespace<TServiceOperation>[];
```

Comment:
instead of providing helper function, i put the namespaces hierarchy info into the current `SdkPackage` type and introduced another `SdkNamepace` type to wrapper the namespace contents.